scipy/optimize/tests/test_bracket.py

```diff
@@ -157,76 +168,85 @@ def f(*args, **kwargs):
-            assert_allclose(res_attr.ravel(), ref_attr)
-            assert_equal(res_attr.shape, shape)
+            xp_assert_close(xp_ravel(res_attr, xp=xp),
+                            xp.asarray(ref_attr, dtype=res_attr.dtype),
+                            rtol=1e-5)
+            xp_assert_equal(res_attr.shape, shape)
 
-        assert np.issubdtype(res.success.dtype, np.bool_)
+        assert xp.isdtype(res.success.dtype, "bool")
         if shape:
-            assert np.all(res.success[1:-1])
-        assert np.issubdtype(res.status.dtype, np.integer)
-        assert np.issubdtype(res.nfev.dtype, np.integer)
-        assert np.issubdtype(res.nit.dtype, np.integer)
-        assert_equal(np.max(res.nit), f.f_evals - 2)
-        assert_array_less(res.xl, res.xr)
-        assert_allclose(res.fl, self.f(res.xl, *args))
-        assert_allclose(res.fr, self.f(res.xr, *args))
-
-    def test_flags(self):
+            assert xp.all(res.success[1:-1])
+        assert xp.isdtype(res.status.dtype, "integral")
+        assert xp.isdtype(res.nfev.dtype, "integral")
+        assert xp.isdtype(res.nit.dtype, "integral")
+        assert xp.max(res.nit) == f.f_evals - 2
+        xp_assert_less(res.xl, res.xr)
+        xp_assert_close(res.fl, xp.asarray(self.f(res.xl, *args)))
+        xp_assert_close(res.fr, xp.asarray(self.f(res.xr, *args)))
+
+    def test_flags(self, xp):
         # Test cases that should produce different status flags; show that all
         # can be produced simultaneously.
         def f(xs, js):
             funcs = [lambda x: x - 1.5,
                      lambda x: x - 1000,
                      lambda x: x - 1000,
-                     lambda x: np.nan,
+                     lambda x: xp.nan,
                      lambda x: x]
 
             return [funcs[j](x) for x, j in zip(xs, js)]
 
-        args = (np.arange(5, dtype=np.int64),)
+        args = (xp.arange(5, dtype=xp.int64),)
         res = _bracket_root(f,
-                            xl0=[-1, -1, -1, -1, 4],
+                            xl0=xp.asarray([-1., -1., -1., -1., 4.]),
                             xr0=[1, 1, 1, 1, -4],
-                            xmin=[-np.inf, -1, -np.inf, -np.inf, 6],
-                            xmax=[np.inf, 1, np.inf, np.inf, 2],
+                            xmin=[-xp.inf, -1, -xp.inf, -xp.inf, 6],
+                            xmax=[xp.inf, 1, xp.inf, xp.inf, 2],
                             args=args, maxiter=3)
 
-        ref_flags = np.array([eim._ECONVERGED,
-                              _ELIMITS,
-                              eim._ECONVERR,
-                              eim._EVALUEERR,
-                              eim._EINPUTERR])
+        ref_flags = xp.asarray([eim._ECONVERGED,
+                                _ELIMITS,
+                                eim._ECONVERR,
+                                eim._EVALUEERR,
+                                eim._EINPUTERR],
+                               dtype=res.status.dtype)
```

Comment:
```suggestion
                               dtype=xp.int32)
```
I do want to keep the type consistent.